analyzer/evmnfts/evm_nfts.go

```diff
@@ -0,0 +1,127 @@
+package evmnfts
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+
+	"github.com/oasisprotocol/nexus/analyzer"
+	"github.com/oasisprotocol/nexus/analyzer/evmnfts/ipfsclient"
+	"github.com/oasisprotocol/nexus/analyzer/item"
+	"github.com/oasisprotocol/nexus/analyzer/queries"
+	"github.com/oasisprotocol/nexus/analyzer/runtime/evm"
+	"github.com/oasisprotocol/nexus/common"
+	"github.com/oasisprotocol/nexus/config"
+	"github.com/oasisprotocol/nexus/log"
+	"github.com/oasisprotocol/nexus/storage"
+	"github.com/oasisprotocol/nexus/storage/client"
+	"github.com/oasisprotocol/nexus/storage/oasis/nodeapi"
+)
+
+const (
+	evmNFTsAnalyzerPrefix = "evm_nfts_"
+)
+
+type processor struct {
+	runtime    common.Runtime
+	source     nodeapi.RuntimeApiLite
+	ipfsClient ipfsclient.Client
+	target     storage.TargetStorage
+	logger     *log.Logger
+}
+
+var _ item.ItemProcessor[*StaleNFT] = (*processor)(nil)
+
+func NewAnalyzer(
+	runtime common.Runtime,
+	cfg config.ItemBasedAnalyzerConfig,
+	sourceClient nodeapi.RuntimeApiLite,
+	ipfsClient ipfsclient.Client,
+	target storage.TargetStorage,
+	logger *log.Logger,
+) (analyzer.Analyzer, error) {
+	logger = logger.With("analyzer", evmNFTsAnalyzerPrefix+runtime)
+	p := &processor{
+		runtime:    runtime,
+		source:     sourceClient,
+		ipfsClient: ipfsClient,
+		target:     target,
+		logger:     logger,
+	}
+	return item.NewAnalyzer[*StaleNFT](
+		evmNFTsAnalyzerPrefix+string(runtime),
+		cfg,
+		p,
+		target,
+		logger,
+	)
+}
+
+type StaleNFT struct {
+	Addr                  string
+	ID                    *big.Int
```

Comment:
Why not use common.BigInt here? The main point of introducing our own BigInt was that it interops more smoothly with pgx and other (de)serializations. Also, doesn't have funky pointer semantics.

Reply:
lately I've been using a lot more go-ethereum types, which it uses *big.Int for its large integers. common.BigInt does come in handy when scanning from the database though.

Reply:
Yeah looking at the DB code was what prompted this comment.
I think this is the right layer to convert from whatever types go-ethereum gives us to the types we prefer to use internally, if the two are different.
Obviously a minor comment though, your call in the end, I won't cry either way.

Reply:
should we just say it? do you want us to prefer using common.BigInt and converting to *big.Int locally where needed?

Reply:
oh. Not sure what "should we just say it" means exactly, but yes, I do prefer what you describe. I think it makes for more consistent code and smoother interop between various parts of the indexer. (EVM work that often needs *big.Int because of go-ethereum is an unfortunate exception.)
So inasmuch as I have decision power, I choose to wield it in that direction :)

Reply:
k going with common.BigInt here :tipping_hand_person: